docs/design/HealthEndpoint.md

```diff
@@ -0,0 +1,417 @@
+
+**Important Point to Note**
+
+In case of Development and Production mode of deployment for DAB, we have certain changes in role permissions. The below table shows cases whether health checks are run if roles are not configured, allowed, and not allowed.
+
+| Role Membership | Health Check
+| - | - |
+|{not configured} | <table><tr><th>Runtime</th><th>Run Checks</th></tr><tr><td>Development</td><td>Yes</td></tr><tr><td>Production</td><td>No (403 Forbidden) </td></tr></table> |
+| Allowed | <table><tr><th>Runtime</th><th>Run Checks</th></tr><tr><td>Development</td><td>Yes</td></tr><tr><td>Production</td><td>Yes</td></tr></table> |
+| Not Allowed | <table><tr><th>Runtime</th><th>Run Checks</th></tr><tr><td>Development</td><td>No (403 Forbidden)</td></tr><tr><td>Production</td><td>No (403 Forbidden)</td></tr></table> |
+
+### Base URL for DAB
+While running the health checks for entities, we need the BASE URL on which DAB is running. This BASE URL is used to run all datasource, rest and graphql health check queries for our engine. This base URL is identified during runtime by using `httpcontext`. The following command fetches the base URL upon which all health checks would be run subsequently. 
+```
+string path = UriHelper.GetEncodedUrl(httpContext!.Request).Split('?')[0];
+```
+
+## Health Endpoint Report Calculation 
+This section contains details on how the health check details are calculated and Comprehensive Health Check report is formulated at the newly updated endpoint: `/health`.
+
+> Case 1: `runtime.health.enabled` is false/null\
+The endpoint returns a 404 (Not Found) error response when hitting the page as comprehensive health check report is disabled.
+
+> Case2:  `runtime.health.enabled` is true\
+The endpoint returns the comprehensive health check report after executing the checks for data-source and entities.
+
+### Configuration Parameters
+The Health Report also contains a section for the configuration summary for the health checks. It provides information about various parameters being enabled or disabled on the global runtime scale. 
+It includes the following
++ Http
++ Https
++ Rest
++ GraphQL
++ Telemetry
++ Caching
++ Mode of Deployment
+
+> NOTE that BASIC health does NOT include configuration.
+
+### Application Name
+The application name created by DAB that is appended to data source connection string; is a rich string that exposes a lot of internal stuff. This would be enhanced in the future, hence use the `GetDataApiBuilderUserAgent()` function from the `ProductInfo` class to fetch this for the health report. 
+
+### DataSource Health Check
+DAB Config has two parameters in the `data-source.health` config section. (Enabled and threshold_ms)\
```

Comment:
nit: two parameters > two settings

Just a reminder, a `parameter` is a variable listed in a method’s signature, used to pass data into the method.